src/langchain-agent-auth.mdx

```diff
@@ -0,0 +1,115 @@
+
+The system consists of:
+- **Authentication Server**: Manages OAuth flows and stores credentials securely
+- **Client SDK**: Python library for easy integration with agents
+
+## Installation
+
+Install directly from the wheel file:
+
+For use with LangGraph:
+```bash
+pip install https://docs.langchain.com/langchain_auth-0.1.0-py3-none-any.whl[langgraph]
+```
+
+For use without LangGraph:
+```bash
+pip install https://docs.langchain.com/langchain_auth-0.1.0-py3-none-any.whl
+```
+
+
+## Quick Start
+
+### 1. Initialize the Client
+
+```python
+from langchain_auth import Client
+
+client = Client(api_key="your-langsmith-api-key")
+```
+
+### 2. Set Up OAuth Providers (one time)
+
+Before agents can authenticate, you need to configure an OAuth provider:
+
+```python
+# Create GitHub provider
+github_provider = await client.create_oauth_provider(
+    provider_id="github",
+    name="GitHub",
+    client_id="your-github-client-id",
+    client_secret="your-github-client-secret",
+    auth_url="https://github.com/login/oauth/authorize",
+    token_url="https://github.com/login/oauth/access_token",
+    user_info_url="https://api.github.com/user",
+    scopes=["repo"] # Provide one or more scopes you would like to be available for this provider
+)
+```
+
+### 3. Authenticate from An Agent
+
+The client `authenticate()` API is used to get OAuth tokens from pre-configured providers. On the first call, it takes the caller through an OAuth 2.0 auth flow.
+
+#### In LangGraph Context
+
+By default, tokens will be scoped to the calling agent using the Assistant ID parameter.
+
+```python
+auth_result = await client.authenticate(
+    provider="github",
+    scopes=["repo"],
+    user_id="your_user_id" # Any unique identifier to scope this token to the human caller
+)
+
+# Or if you'd like a token that can be used by any agent, set agent_scoped=False
+auth_result = await client.authenticate(
+    provider="github",
+    scopes=["repo"],
+    user_id="your_user_id",
+    agent_scoped=False
+)
+```
+
+During execution, if authentication is required, the SDK will throw an [interrupt](https://langchain-ai.github.io/langgraph/how-tos/human_in_the_loop/add-human-in-the-loop/#pause-using-interrupt). The agent execution pauses and presents the OAuth URL to the user:
+
+![LangGraph Studio interrupt showing OAuth URL](./images/langgraph-auth-interrupt.png)
+
+After the user completes OAuth authentication and we receive the callback from the provider, they will see the auth success page:
+
+![GitHub OAuth success page](./images/github-auth-success.png)
+
+The agent resumes execution from the point it left off. The token can now be used for any API calls. We store and refresh OAuth tokens on our side, meaning any future uses of this service and scope from this user and agent will not require an OAuth flow.
```

Comment:
```suggestion
The agent then resumes execution from the point it left off at, and the token can be used for any API calls. We store and refresh OAuth tokens on our side so that future uses of the service by either the user or agent don't require an OAuth flow.
```

That may not be the best suggestion but the last sentence was a little hard to follow so I tried to rework it a little. If what I suggested was technically inaccurate, please feel free to ignore or edit it.